doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,90 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically. For example, solving $x^2 = 4$ yields $x \in \{-2,2\}$.
+
+There are two high-level functions to solve equations, [`solve`](#) and [`solveset`](#). Here are their advantages and disadvantages: ... *table?*
+
+## Using {func}`~.solve`
+
+{func}`~.solve`
+- is best for solving equations that...
+- produces outputs in the format of a Python list.
+
+You can solve an equation using {func}`~.solve` in several ways.
+
+1. Use the fact that any expression not in an `Eq` (equation) is automatically assumed to equal 0 by the solving functions. You can rearrange the equation $x^2 = 4$ to $x^2 - 4 = 0$, and {func}`~.solve` that expression. This approach is convenient if you are interactively solving an equation which can easily be rearranged to $expression = 0$.
+
+```
+>>> from sympy import solve
+>>> from sympy.abc import x
+>>> solve(x**2 - 4)
+[-2, 2]
+```
+
+2. Put your equation into `Eq` form, then apply {func}`~.solve` to the `Eq`. This approach is convenient if you are interactively solving an equation which cannot easily be rearranged to $expression = 0$.
+
+```
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x
+>>> eqn = Eq(x**2, 4)
+>>> solve(eqn)
+[-2, 2]
+```
+
+3. Parse a string representing the equation into a form that SymPy can understand, then apply {func}`~.solve` to the parsed expression. 
+
+*But this doesn't work for equality; does for inequality. Is there some way to do this for equality?*
+
+```
+>>> from sympy import solve, parse_expr
+>>> expr = "x**2 == 4"
+>>> parsed = parse_expr(expr)
+>>> solve(parsed)
+[-2, 2]
+```
+
+## Using {func}`~.solveset`
+
+{func}`~.solveset`
+- is best for solving equations that...
+- produces outputs in the format of [SymPy mathematical Sets](https://docs.sympy.org/dev/modules/sets.html?highlight=sets#module-sympy.sets.sets) rather than [Python sets](https://docs.python.org/3/library/stdtypes.html#set)
+- can return infinitely many solutions
+- clearly separates the complex and real domains
+
+```
+>>> from sympy import solveset
+>>> from sympy.abc import x
+>>> solveset(x**2 - 4)
+FiniteSet(-2, 2)
+```
+
+## Not all equations can be solved algebraically
+
+Some equations have no algebraic solution, in which case SymPy may return an empty set:
+
+```
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x
+>>> eqn = Eq(3 * x - 7, 3 * x + 2)
+>>> solve(eqn)
+[]
```

Comment:
The phrase "algebraic solution" is misleading. Here this equation does not have any solution full stop.

Other equations have solutions that cannot be expressed algebraically (symbolically) e.g.:
```python
In [12]: solve(cos(x)-x)
---------------------------------------------------------------------------
NotImplementedError
```
This equation has solutions (e.g. check `plot(cos(x), x)` to see the real one). We can find those numerically:
```python
In [14]: nsolve(cos(x)-x, x, 0)
Out[14]: 0.739085133215161
```
It is not however (as far as I know) possible to express the solution of this equation using any ordinary symbolic expression that anyone would recognise. There are different levels of what constitutes a solution expression:
https://en.wikipedia.org/wiki/Closed-form_expression#Analytic_expression

The `solve` function generally aims to compute solutions as explicit analytic expressions.